### ISSUE TITLE
Train Accuracy in evaluation mode is decreased by every epoch

### ISSUE THREAD

**Issue by hoangthangta** [crux82/ganbert-pytorch#14]
I did the same evaluation after every epoch for test set, but now is for **train set**.  However the training accuracy is decreased while the train loss is increased, whap happened here?

`# ========================================
        #     TEST ON THE TRAINING DATASET
        # ========================================
        # After the completion of each training epoch, measure our performance on
        # our training set.
        
        print("")
        print("Running Training...")

        t1 = time.time()
        
        # Tracking variables for train set
        total_train_loss = 0
        all_train_preds = []
        all_train_labels_ids = []
        nll_train_loss = torch.nn.CrossEntropyLoss(ignore_index=-1) #loss

        # Evaluate data for one epoch
        for batch in train_dataloader:
            
            # Unpack this training batch from our dataloader. 
            b_input_ids = batch[0].to(device)
            b_input_mask = batch[1].to(device)
            b_labels = batch[2].to(device)
            
            # Tell pytorch not to bother with constructing the compute graph during
            # the forward pass, since this is only needed for backprop (training).
            with torch.no_grad():        
                model_outputs = transformer(b_input_ids, attention_mask=b_input_mask)
                hidden_states = model_outputs[-1]
                _, logits, probs = discriminator(hidden_states)
                ###log_probs = F.log_softmax(probs[:,1:], dim=-1)
                filtered_logits = logits[:,0:-1]
                # Accumulate the test loss.
                total_train_loss += nll_train_loss(filtered_logits, b_labels)
                
            # Accumulate the predictions and the input labels
            _, preds = torch.max(filtered_logits, 1)
            all_train_preds += preds.detach().cpu()
            all_train_labels_ids += b_labels.detach().cpu()

        # Report the final accuracy for this validation run.
        all_train_preds = torch.stack(all_train_preds).numpy()
        all_train_labels_ids = torch.stack(all_train_labels_ids).numpy()
        train_accuracy = np.sum(all_train_preds == all_train_labels_ids) / len(all_train_preds)
        print("  Train Accuracy: {0:.3f}".format(train_accuracy))

        # Calculate the average loss over all of the batches.
        avg_train_loss = total_train_loss / len(train_dataloader)
        avg_train_loss = avg_train_loss.item()
        
        # Measure how long the validation run took.
        train_time = format_time(time.time() - t1)
        
        print("  Train loss: {0:.3f}".format(avg_train_loss))
        print("  Train took: {:}".format(train_time))`

**Comment by hoangthangta** [comment]
I use learning rate = 1e-6 and it works.
